### PR TITLE
Add dictionaries for almost each language on Wiktionary

### PR DESCRIPTION
**Pull request by Vuizur**
Hello everyone,

I used my [dictionary creation project](https://github.com/Vuizur/ebook_dictionary_creator) based on the [Wiktextract ](https://github.com/tatuylonen/wiktextract) data to create dictionaries in almost every language available on Wiktionary. I don't have enough disk space for Android Studio, so I couldn't test it directly, but I hope the dictionaries.lua file is correctly formatted.

One question remaining is the "relevance criterium" for a language to be included so that super exotic languages don't flood the language selection screen. Currently the data is limited to languages that have more than 500 definitions, but it maybe would be good to raise this - for example to include only dictionaries with more than 1000 entries. And one should probably also remove all Proto-languages because there aren't any texts in them as far as I know. What do you think?

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9767)

### CODE REVIEW

**Comment by Frenzie** [comment]
If I understand correctly this is English Wiktionary split up into sections?

**Comment by Vuizur** [comment]
Yes, this is the English Wiktionary with one dictionary for each language. What is "unique" about it compared to other scraped versions of Wiktionary is that it has pretty good inflections due to the huge effort the wiktextract guys made to process the inflection tables for all languages. And I tried to resolve redirects that sometimes go over multiple entries and did some postprocessing (one could probably still do more if one knew the exact details about each language).

At the end each dictionary has the part of speech in italics and a list of all definitions, so no ugly Wikitext or useless entries like "genitive of ..." instead of the proper definition.

File: frontend/ui/data/dictionaries.lua [hunk 1]
```diff
@@ -1195,6 +1195,2750 @@ local dictionaries = {
         license = "",
         url = "https://gitlab.com/avsej/dicts-stardict-form-xdxf/raw/d636cc5e8d4a47e22ac7466f4af6d435a8a3f650/002c/stardict-comn_sdict05_brokg-2.4.2.tar.gz"
     },
+    {
+        name = "Abkhaz-English Wiktionary dictionary",
```
Comment (Frenzie) [inline frontend/ui/data/dictionaries.lua:1199]: For the name I'd just keep it as Wiktionary for all of them:
```suggestion
        name = "Abkhaz-English Wiktionary",
```
After all, this is displayed in the dictionary download dialog.

File: frontend/ui/data/dictionaries.lua [hunk 2]
```diff
@@ -1195,6 +1195,2750 @@ local dictionaries = {
...
+        url = "https://github.com/Vuizur/Wiktionary-Dictionaries/raw/master/Proto-Finnic-English%20Wiktionary%20dictionary%20stardict.tar.gz",
+    },
+    {
+        name = "Proto-Germanic-English Wiktionary dictionary",
+        lang_in = "Proto-Germanic",
+        lang_out = "English",
+        entries = 4798,
+        license = "Dual-licensed under CC-BY-SA 3.0 and GFDL",
+        url = "https://github.com/Vuizur/Wiktionary-Dictionaries/raw/master/Proto-Germanic-English%20Wiktionary%20dictionary%20stardict.tar.gz",
+    },
```
Comment (Frenzie) [inline frontend/ui/data/dictionaries.lua:3165]: I use these types of entries regularly actually, though perhaps not on my Kobo. I tend to prefer to use a second device or a paper dictionary unless I'm on the train or something.

**Comment by Vuizur** [comment]
Thanks for the comments! Should I drop all languages with under 1000 entries? Without filtering I have 343 languages, after filtering using this criterium remain 189 languages.

**Comment by Frenzie** [comment]
@NiLuJe How do you mean exactly?
image

**Comment by NiLuJe** [comment]
Just that, yeah ;). It's at the top of the entry in the other conversions, and they're not in these (again, last I checked, unless I was really unlucky).

**Comment by Frenzie** [comment]
> Thanks for the comments! Should I drop all languages with under 1000 entries? Without filtering I have 343 languages, after filtering using this criterium remain 189 languages.

I suppose that would make sense, yes, unless we somehow improve the interface.

**Comment by Vuizur** [comment]
The dictionaries now have at least 1000 entries. It is a good point about the IPA, for the languages I learned recently I never needed it, but for English with its illogical pronunciation it definitely is important :). So I will probably add it when I have time again.

File: frontend/ui/data/dictionaries.lua [hunk 3]
```diff
@@ -1195,6 +1195,1518 @@ local dictionaries = {
         license = "",
         url = "https://gitlab.com/avsej/dicts-stardict-form-xdxf/raw/d636cc5e8d4a47e22ac7466f4af6d435a8a3f650/002c/stardict-comn_sdict05_brokg-2.4.2.tar.gz"
     },
+    {
+        name = "Adyghe-English Wiktionary",
+        lang_in = "Adyghe",
+        lang_out = "English",
+        entries = 4576,
+        license = "Dual-licensed under CC-BY-SA 3.0 and GFDL",
+        url = "https://github.com/Vuizur/Wiktionary-Dictionaries/raw/master/Adyghe-English%20Wiktionary%20dictionary%20stardict.tar.gz"
```
Comment (Frenzie) [inline frontend/ui/data/dictionaries.lua:1204]: Please add the trailing comma back. Nothing wrong about it; our style asks for it. ;-)

**Comment by Frenzie** [comment]
Thanks!

**Comment by gbyl** [comment]
@Vuizur Thank you for this outstanding contribution! This seems much better than the alternative I explored at one point in https://github.com/koreader/koreader/issues/7198. Could you please add a version of the English Wiktionary stardict that includes the etymology section?

**Comment by Vuizur** [comment]
@gbyl Thanks. I have added etymologies to my TODO list, but can't promise anything (depends on how much time I will have).

**Comment by M0Rf30** [comment]
Hi everyone. I don't know if out of topic, but I want to reference this project
https://github.com/BoboTiG/ebook-reader-dict
I'm using its resources on koreader with success

**Comment by M0Rf30** [comment]
Hi everyone. I don't know if out of topic, but I want to reference this project
https://github.com/BoboTiG/ebook-reader-dict
I'm using its resources on koreader with success

**Pull request merged**